packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,317 @@
+    }
+    if (!signatureResponse.ok) {
+      return Err(new FetchError(new Error(signatureResponse.statusText)))
+    }
+
+    const [dataBody, signatureBody] = await Promise.all([
+      responseBuffer(dataResponse),
+      responseBuffer(signatureResponse),
+    ])
+
+    const body = Buffer.from(dataBody)
+    const signature = ensureLeading0x(Buffer.from(signatureBody).toString('hex'))
+
+    const toParse = type ? JSON.parse(body.toString()) : body
+    const typedData = await buildEIP712TypedData(this, dataPath, toParse, type)
+    const guessedSigner = recoverEIP712TypedDataSigner(typedData, signature)
+    if (eqAddress(guessedSigner, account)) {
+      return Ok(body)
+    }
+
+    return Err(new InvalidSignature())
+  }
+}
+
+// requires that the DEK has already been registered
+export function* uploadProfileInfo() {
+  const isAlreadyUploaded = yield select(isProfileUploadedSelector)
+  if (isAlreadyUploaded) {
+    return
+  }
+  try {
+    yield call(unlockDEK, true)
+    // yield call(addMetadataClaim)
+    yield call(uploadNameAndPicture)
+
+    yield put(profileUploaded())
+  } catch (e) {
+    Logger.error(TAG + '@uploadProfileInfo', 'Error uploading profile', e)
+  }
+}
+
+// TODO: make metadata claim when registering account info, and use fetched metadata url when reading data
+
+// export function* addMetadataClaim() {
+//   try {
+//     const contractKit = yield call(getContractKit)
+//     const account = yield select(currentAccountSelector)
+//     const metadata = IdentityMetadataWrapper.fromEmpty(account)
+//     yield call(
+//       [metadata, 'addClaim'],
+//       createStorageClaim(BUCKET_URL),
+//       NativeSigner(contractKit.web3.eth.sign, account)
+//     )
+//     Logger.info(TAG + '@addMetadataClaim' + 'created storage claim on chain')
+//     yield call(writeToGCPBucket, metadata.toString(), `${account}/metadata.json`)
+//     Logger.info(TAG + '@addMetadataClaim' + 'uploaded metadata.json')
+//     const accountsWrapper: AccountsWrapper = yield call([
+//       contractKit.contracts,
+//       contractKit.contracts.getAccounts,
+//     ])
+//     const setAccountTx = accountsWrapper.setMetadataURL(`${BUCKET_URL}${account}/metadata.json`)
+//     const context = newTransactionContext(TAG, 'Set metadata URL')
+//     yield call(sendTransaction, setAccountTx.txo, account, context)
+//     Logger.info(TAG + '@addMetadataClaim' + 'set metadata URL')
+//   } catch (error) {
+//     Logger.error(`${TAG}/addMetadataClaim`, 'Could not add metadata claim', error)
+//     throw error
+//   }
+// }
+
+export function* uploadNameAndPicture() {
+  const contractKit = yield call(getContractKit)
+  const account = yield select(currentAccountSelector)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const name = yield select(nameSelector)
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'write'], { name }, [])
```

Comment:
Do we want to consider returning early if the nameAccessor write fails? Otherwise we lose the error if we also upload a profilePicture and that fails too.